src/OrchardCore/OrchardCore.Liquid.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using Fluid;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace OrchardCore.Liquid
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddLiquidFilter<TLiquidFilter>(this IServiceCollection services, string name) where TLiquidFilter : class, ILiquidFilter
+        {
+            // services.Configure<TemplateOptions>(o => o.Filters.AddFilter(name, new LiquidFilterResolver(typeof(T)).ResolveThenProcessAsync));
+            services.Configure<TemplateOptions>(o => o.Filters.AddFilter(name, new LiquidFilterResolver<TLiquidFilter>().ResolveThenProcessAsync));
+            services.AddTransient<TLiquidFilter>();
```

Comment:
I've added them as `Transient`. They could equally be `Scoped` I think.

Need to decide on what is best there.

Went with `Transient` as @jtkech has confirmed that all the `TagHelpers` are always transient, so trying to keep some consistency. However `TagHelpers` need to be transient because they do bindings, whereas liquid filters, don't bind. So maybe scoped is advantageous.

Thoughts @jtkech ?

Reply:
@deanmarcussen 

As liquid filters, i think it is better to register them as scoped services so that the services are injected once per scope, unless for some if they need to hold something that would need to be different per instance, but i don't think so, e.g. as the view context for a given tag helper (if it has this property), and as you said because of they bound data.

Reply:
@jtkech cool. I debated with myself about this.

On the one hand, liquid template manager is scoped. On the other hand, it maybe that during a request 3 different templates are rendering. (more on the admin tbf, but also with liquid parts or the liquid filter, where inside a liquid page, you might render another liquid template in the same request). 

So scoped is fine, so long as no one tries to hold scoped values in them - but where we have done recursion checking, we used a scoped helper anyway, so it's abstracted out. 

All good to go with scoped.

Reply:
@deanmarcussen 

Okay, so in `AddLiquidFilter()` we can use `AddScoped<>()`

But in `AddTagHelpers()` we still need `AddTransient<>()`

I will do the change

Reply:
Cool. I just tweaked it to remove my commented out code.

Reply:
I hesitated to do it myself ;)